src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -724,7 +724,9 @@ private function addNewInstance($id, Definition $definition, $return, $instantia
             }
 
             if (null !== $definition->getFactoryService()) {
-                return sprintf("        $return{$instantiation}%s->%s(%s);\n", $this->getServiceCall($definition->getFactoryService()), $definition->getFactoryMethod(), implode(', ', $arguments));
+                $service = $this->dumpValue($definition->getFactoryService());
+
+                return sprintf("        $return{$instantiation}%s->%s(%s);\n", 0 === strpos($service, '$') ? sprintf('$this->get(%s)', $service) : $this->getServiceCall($definition->getFactoryService()), $definition->getFactoryMethod(), implode(', ', $arguments));
```

Comment:
IIRC, it's because it's supported by the new way of defining factories. But you're right, it does not really make sense. I'm going to drop this.